File: db4on/reference/db4o-reference-chapters/src/com/db4odoc/f1/refactoring/RefactoringExample.cs
/* Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com */
using System;
using System.IO;
using com.db4o;
using com.db4o.query;

namespace com.db4odoc.f1.refactoring
{
	public class RefactoringExample
	{
		public readonly static string YapFileName = "formula1.yap";
	
		public static void Main(string[] args) 
		{
			System.Console.WriteLine("Correct sequence of actions: ");
			SetObjects();
			CheckDB();	
			ChangeClass();
			SetNewObjects();
			RetrievePilotNew();

			/*System.Console.WriteLine("Incorrect sequence of actions: ");
			SetObjects();
			CheckDB();	
			SetNewObjects();
			ChangeClass();
			RetrievePilotNew();*/
		}
		// end Main

		public static void SetObjects()
		{
			File.Delete(YapFileName);
			ObjectContainer oc = Db4o.OpenFile(YapFileName);
			try 
			{
				Pilot pilot = new Pilot("Rubens Barrichello");
				oc.Set(pilot);
				pilot = new Pilot("Michael Schumacher");
				oc.Set(pilot);
			} 
			finally 
			{
				oc.Close();
			}
		}
		// end SetObjects

		public static void CheckDB()
		{
			ObjectContainer oc = Db4o.OpenFile(YapFileName);
			try 
			{
				ObjectSet result = oc.Get(typeof(Pilot));
				for (int i=0; i< result.Size();i++)
				{
					Pilot pilot = (Pilot)result[i];
					System.Console.WriteLine("Pilot="+ pilot);
				}
			} 
			finally 
			{
				oc.Close();
			}
		}
		// end CheckDB

		public static void SetNewObjects()
		{
			ObjectContainer oc = Db4o.OpenFile(YapFileName);
			try 
			{
				PilotNew pilot = new PilotNew("Rubens Barrichello",99);
				oc.Set(pilot);
				pilot = new PilotNew("Michael Schumacher",100);
				oc.Set(pilot);
			} 
			finally 
			{
				oc.Close();
			}
		}
		// end SetNewObjects

		public static void ChangeClass()
		{
			Db4o.Configure().ObjectClass(typeof(Pilot)).Rename("com.db4odoc.f1.refactoring.PilotNew, db4o-reference-chapters");
			Db4o.Configure().ObjectClass(typeof(PilotNew)).ObjectField("_name").Rename("_identity");
			ObjectContainer oc = Db4o.OpenFile(YapFileName);
			oc.Close();
		}
		// end ChangeClass
	
		public static void RetrievePilotNew()
		{
			ObjectContainer oc = Db4o.OpenFile(YapFileName);
			try 
			{
				Query q = oc.Query();
				q.Constrain(typeof(PilotNew));
				ObjectSet result = q.Execute();
				for (int i=0; i< result.Size();i++)
				{
					PilotNew pilot = (PilotNew)result[i];
					System.Console.WriteLine("Pilot="+ pilot);
				} 
			}
			finally 
			{
				oc.Close();
			}
		}
		// end RetrievePilotNew
	}
}

File: db4on/reference/vb/db4o-reference-chapters/src/com/db4odoc/f1/refactoring/RefactoringExample.vb
' Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com 
Imports System
Imports System.IO
Imports com.db4o
Imports com.db4o.query

Namespace com.db4odoc.f1.refactoring
    Public Class RefactoringExample
        Public Shared ReadOnly YapFileName As String = "formula1.yap"

        Public Shared Sub Main(ByVal args() As String)
            System.Console.WriteLine("Correct sequence of actions: ")
            SetObjects()
            CheckDB()
            ChangeClass()
            SetNewObjects()
            RetrievePilotNew()

            '           System.Console.WriteLine("Incorrect sequence of actions: ")
            ' 			SetObjects()
            ' 			CheckDB()	
            ' 			SetNewObjects()
            ' 			ChangeClass()
            ' 			RetrievePilotNew()*/
        End Sub
        ' end Main

        Public Shared Sub SetObjects()
            File.Delete(YapFileName)
            Dim oc As ObjectContainer = Db4o.OpenFile(YapFileName)
            Try
                Dim pilot As Pilot = New Pilot("Rubens Barrichello")
                oc.Set(pilot)
                pilot = New Pilot("Michael Schumacher")
                oc.Set(pilot)
            Finally
                oc.Close()
            End Try
        End Sub
        ' end SetObjects

        Public Shared Sub CheckDB()
            Dim oc As ObjectContainer = Db4o.OpenFile(YapFileName)
            Try
                Dim result As ObjectSet = oc.Get(GetType(Pilot))
                Dim i As Integer
                For i = 0 To result.Size() - 1 Step i + 1
                    Dim pilot As Pilot = CType(result(i), Pilot)
                    System.Console.WriteLine("Pilot=" + pilot.ToString())
                Next
            Finally
                oc.Close()
            End Try
        End Sub
        ' end CheckDB

        Public Shared Sub SetNewObjects()
            Dim oc As ObjectContainer = Db4o.OpenFile(YapFileName)
            Try
                Dim pilot As PilotNew = New PilotNew("Rubens Barrichello", 99)
                oc.Set(pilot)
                pilot = New PilotNew("Michael Schumacher", 100)
                oc.Set(pilot)
            Finally
                oc.Close()
            End Try
        End Sub
        ' end SetNewObjects

        Public Shared Sub ChangeClass()
            Db4o.Configure().ObjectClass(GetType(Pilot)).Rename("com.db4odoc.f1.refactoring.PilotNew,db4o-reference-chapters")
            Db4o.Configure().ObjectClass(GetType(PilotNew)).ObjectField("_name").Rename("_identity")
            Dim oc As ObjectContainer = Db4o.OpenFile(YapFileName)
            oc.Close()
        End Sub
        ' end ChangeClass

        Public Shared Sub RetrievePilotNew()
            Dim oc As ObjectContainer = Db4o.OpenFile(YapFileName)
            Try
                Dim q As Query = oc.Query()
                q.Constrain(GetType(PilotNew))
                Dim result As ObjectSet = q.Execute()
                Dim i As Integer
                For i = 0 To result.Size() - 1 Step i + 1
                    Dim pilot As PilotNew = CType(result(i), PilotNew)
                    System.Console.WriteLine("Pilot=" + pilot.ToString())
                Next
            Finally
                oc.Close()
            End Try
        End Sub
        ' end RetrievePilotNew
    End Class
End Namespace

